examples/platforms/simulation/system.c

```diff
@@ -65,54 +68,97 @@ static void handleSignal(int aSignal)
 
 void otSysInit(int aArgCount, char *aArgVector[])
 {
-    char *   endptr;
     uint32_t speedUpFactor = 1;
-    int      argIndex      = 0;
+#if OPENTHREAD_CONFIG_LOG_OUTPUT == OPENTHREAD_CONFIG_LOG_OUTPUT_FILE
+    char logFileName[128];
+    memset(logFileName, 0, sizeof(logFileName));
+#endif
 
     if (gPlatformPseudoResetWasRequested)
     {
         gPlatformPseudoResetWasRequested = false;
         return;
     }
 
-    if (aArgCount < 2)
+    struct option long_options[] =
     {
-        fprintf(stderr, "Syntax:\n    %s [--sleep-to-tx] NodeId [TimeSpeedUpFactor]\n", aArgVector[0]);
-        exit(EXIT_FAILURE);
-    }
+        {"sleep-to-tx", no_argument, 0, 's'},
+        {"time-speed-up-factor", required_argument, 0, 't'},
+#if OPENTHREAD_CONFIG_LOG_OUTPUT == OPENTHREAD_CONFIG_LOG_OUTPUT_FILE
+        {"log-file", required_argument, 0, 'l'},
+#endif
+    };
 
-    openlog(basename(aArgVector[argIndex++]), LOG_PID, LOG_USER);
-    setlogmask(setlogmask(0) & LOG_UPTO(LOG_NOTICE));
+    while (true)
+    {
+        int c = getopt_long(aArgCount, aArgVector, "", long_options, NULL);
 
-    signal(SIGTERM, &handleSignal);
-    signal(SIGHUP, &handleSignal);
+        if (c == -1)
+        {
+            break;
+        }
+
+        switch (c)
+        {
+        case 's':
+            gRadioCaps |= OT_RADIO_CAPS_SLEEP_TO_TX;
+            break;
+        case 't':
+            speedUpFactor = (uint32_t)strtol(optarg, NULL, 10);
+            break;
+#if OPENTHREAD_CONFIG_LOG_OUTPUT == OPENTHREAD_CONFIG_LOG_OUTPUT_FILE
```

Comment:
I seems we can simply achieve the effect by using `stderr`.
```suggestion
#if OPENTHREAD_CONFIG_LOG_OUTPUT == OPENTHREAD_CONFIG_LOG_OUTPUT_STDERR
```

usage will be:
```
ot-cli-ftd 1 2> /path/to/logging_file
```